### PR TITLE
SF-663 Fix missing questions on overview page

### PR DESCRIPTION
**Pull request by Nateowami**
- This is the fourth time we've worked on this issue in one way or another, with the previous fix being c12d511720b9d31f24228d639c4bb4f59e257f7a
- Also a couple miscellaneous changes

Previously `initTexts` was only called when the route changed (switching projects) or `projectDoc.remoteChanges$` or `questionsQuery.remoteChanges$` provides us new data. It would appear `remoteChanges$` means just that; remote changes. That doesn't really explain why it happens so much more often when using a production build though, and doesn't always happen with non-production builds. I do think `initTexts` is idempotent though, so I'm not really worried about it causing problems.

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/web-xforge/619)

### CODE REVIEW

**Comment by Nateowami** [comment]
I just looked into this, and it appears the way we mock the adding of the question occurs in memory and circumvents the actual problem. Here's the function we use to submit questions:

``` ts
addQuestion(question: Question): void {
  this.realtimeService.addSnapshot<Question>(QUESTIONS_COLLECTION, {
    id: getQuestionDocId('project01', question.dataId),
    data: question
  });
}
```

We already have tests called "should show new question after adding" and "should show new question after adding to a project with no questions". I don't think adding a test that will actually show the problem is actually feasible.

**Comment by Nateowami** [comment]
@irahopkinson Part of the issue is that `initTexts` actually is called in the tests, and has been even before this fix. I think the following code is where the real problem lies:

``` ts
this.dataChangesSub = merge(this.projectDoc.remoteChanges$, this.questionsQuery.remoteChanges$).subscribe(() => {
  this.loadingStarted();
  try {
    this.initTexts();
  } finally {
    this.loadingFinished();
  }
});
```

This is where the call to `initTexts` sometimes happens after a question is added and sometimes doesn't. It appears to me that the problem is that we were only listening to remote changes, and not local changes (I'm assuming that `projectDoc.remoteChanges$` and `questionsQuery.remoteChanges$` are aptly named). Before this commit, there were only two places `initTexts` was called, and the other isn't really relevant to this issue. I would have expected `this.projectDoc` and `this.questionsQuery` to have some property like `changes$` (for remote *or* local changes), but it doesn't appear they do. The ideal fix would be to change the query to trigger for remote or local changes, but since that doesn't seem to be an option, I just added a call to `initTexts` immediately after a question is added locally.

**Comment by Nateowami** [comment]
I just looked through the history of changes:

\#| PR   | Issue                                        
--|------|----------------------------------------------
1 | #430 | [SF-663](https://jira.sil.org/browse/SF-663) The question does not appear on the overview page.
2 | #516 | [SF-752](https://jira.sil.org/browse/SF-752) The book content does not disappear after archiving a question.
3 | #535 | [SF-777](https://jira.sil.org/browse/SF-777) Duplicate question appears when adding the second question

I haven't dug through everything fully, since it involves a lot of realtime query stuff that I would get lost in, but I think I have a fairly good idea what happened:
1. Our realtime query code didn't await everything it needed to await when a doc was updated on the front end (Damien's change, #516 is called "Await local query updates when a doc is updated"). This caused multiple problems, but in our case it can be seen in the line `await this.questionDialogService.questionDialog(data, questionDoc);`. The lines following then assume that the question has been added to the collection of questions in memory (`this.projectDoc.data.texts`). However, because some things were not awaited within the realtime query logic, it actually returns *before* the question is added to the collection.
2. The following lines used to come after the call to the question dialog service, and they assumed the question had been added to the collection:
``` ts
this.loadingStarted();
try {
  this.initTexts();
} finally {
  this.loadingFinished();
}
```
When `initTexts` is called, it reads from `this.projectDoc.data.texts`, which still has the old data. Most likely the reason we didn't notice this issue is because the query for remote changes sometimes/usually fires after the data takes a round trip from the server (this is only a hypothesis), which causes a call to `initTexts`.
3. The fix for this (#430) threw out the call to `initTexts` and just directly added the question to the list of questions to be shown.
4. The fix for the awaiting problem in #516 caused the question to be added to the collection when it should be, in the line `await this.questionDialogService.questionDialog(data, questionDoc);`. If the remote changes fired immediately, then `initTexts` would be called *before* the question was pushed onto the list of questions to show. Since the addition of that single question is not an idempotent process like `initTexts` is, if it occurs after `initTexts`, we can end up with two copies of the same question. That the remote changes were firing is only a hypothesis, but I don't think anything else could explain the behavior that was occurring.
5. I came along in #535 and threw out the code that adds the single question to the list of questions to display. While that fixed the duplicate question, it also meant that the question list would only update when the remote query fired. (This entire explanation is predicated on the idea that the addition of a question sometimes causes the remote changes listener to fire, and sometimes does not, with no hypothesis as to why that may be).
6. In this PR I've added back the call to `initTexts`. However, it would probably be better to revert it to how it originally was:
``` ts
this.loadingStarted();
try {
  this.initTexts();
} finally {
  this.loadingFinished();
}
```
In summary, this code was right all along, was incorrectly fixed, then I incorrectly reverted it by deleting what it was changed to. The real issue was in the realtime query system, and the problem was masked by inconsistent/unpredictable behavior from the queries.

**Comment by Nateowami** [comment]
Something reassuring I've found testing these latest changes:
- If I go back a few commits and kill the server, I can add multiple questions and none of them show up. When I don't kill the server the tests results have been inconsistent and adding a second question always caused them to all show up. I think this helps confirm that a server response is triggering the showing of the questions.
- After this fix, if I kill the server, the questions show up on the page, despite being completely disconnected from the server.
